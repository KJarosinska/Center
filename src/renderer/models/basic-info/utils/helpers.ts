import { SimCard } from "Renderer/models/basic-info/interfaces"

export const getActiveNetworkFromSim = (simCards: SimCard[] = []) => {
  const activeSimCard = simCards.filter(simCard => simCard.active)
  if (activeSimCard.length === 0) {
    return "No connection"
  }
  return activeSimCard[0].network
}
