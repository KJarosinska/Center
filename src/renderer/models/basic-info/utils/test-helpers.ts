import { IpcRequest } from "Common/requests/ipc-request.enum"

export const commonCalls = {
  [IpcRequest.GetDeviceInfo]: Promise.resolve({
    name: "Ziemniaczek",
    modelName: "U12300000",
    modelNumber: "A1239999",
    serilaNumber: "a-b-3d",
    osVersion: "0.123v",
    osUpdateDate: "12-12-2003",
  }),
  [IpcRequest.GetNetworkInfo]: Promise.resolve({
    simCards: [
      {
        active: true,
        carrier: "AAAAAAAAAAAA",
        iccid: 1234,
        imei: 5678,
        meid: 8765,
        network: "Y-Mobile",
        networkLevel: 0.5,
        number: 12345678,
        seid: "1234",
        slot: 1,
      },
      {
        active: false,
        carrier: "BBBBBBBBBBBB",
        iccid: 412,
        imei: 42,
        meid: 1410,
        network: "X-Mobile",
        networkLevel: 0.69,
        number: 7001234523,
        seid: "x123",
        slot: 2,
      },
    ],
  }),
  [IpcRequest.GetStorageInfo]: Promise.resolve({
    capacity: 9001,
    available: 99999999999999,
    categories: [
      { label: "music", filesCount: 1233333, size: 999999999 },
      { label: "storage", filesCount: 100000, size: 999999999 },
    ],
  }),
  [IpcRequest.GetBatteryInfo]: Promise.resolve({
    level: 9001,
    charging: false,
    maximumCapacity: 99999,
  }),
  [IpcRequest.GetBackupsInfo]: Promise.resolve({
    backups: [
      {
        createdAt: "20-11-15T07:35:01.562Z20",
        size: 99999,
      },
      {
        createdAt: "20-01-30T07:35:01.562Z20",
        size: 1234567,
      },
    ],
  }),
}
