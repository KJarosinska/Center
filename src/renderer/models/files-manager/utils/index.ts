import { FilesManagerState } from "Renderer/models/files-manager/files-manager.interface"

export const prepareDataForStackedBarChart = (data: FilesManagerState) => {
  return data.memoryData.map(el => {
    const { filesType, filesAmount, occupiedMemory, ...rest } = el
    return {
      value: occupiedMemory,
      ...rest,
    }
  })
}
