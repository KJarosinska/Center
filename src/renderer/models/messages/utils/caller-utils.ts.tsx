import { Author } from "Renderer/models/messages/messages.interface"

const removeDecoratorsFromPhoneNumber = (string = ""): string => {
  return string.split(" ").join("").replace("+", "")
}

export interface CallerSearchParams extends Record<string, string> {
  id: string
  phoneNumber: string
}

export const isCallerMatchingSearchParams = (
  caller: Author,
  params: CallerSearchParams
): boolean => {
  return (
    params.id === caller.id &&
    removeDecoratorsFromPhoneNumber(caller.primaryPhoneNumber) ===
      removeDecoratorsFromPhoneNumber(params.phoneNumber)
  )
}
