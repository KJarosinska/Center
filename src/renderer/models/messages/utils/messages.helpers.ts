import { Contact } from "Renderer/models/phone/phone.typings"
import { Topic } from "Renderer/models/messages/messages.interface"

export type ContactsCollection = Record<string, Contact>
export type Getter = (
  id: string,
  collection: ContactsCollection
) => Contact | undefined

export const getContactsAsMap = (contacts: Contact[]) => {
  return contacts.reduce((acc: ContactsCollection, item: Contact) => {
    return {
      ...acc,
      [item.id]: item,
    }
  }, {})
}

export const getContactDetails = (
  id: string,
  collection: ContactsCollection
): Contact | undefined => {
  if (id in collection) {
    return collection[id]
  }

  return undefined
}

export const expandTopic = (
  topic: Topic,
  collection: ContactsCollection,
  getter: Getter
) => {
  const { messages } = topic

  return {
    ...topic,
    messages: messages.map((msg) => {
      const author = getter(msg.author.id, collection)

      if (author) {
        return {
          ...msg,
          author,
        }
      }

      return msg
    }),
  }
}

export const createFullMessagesCollection = (state: any) => {
  const {
    messages: { topics },
    phone: { contacts: baseContacts },
  } = state

  const contacts = getContactsAsMap(baseContacts)

  return topics.map((topic: Topic) => {
    const { id } = topic.caller
    const caller = getContactDetails(id, contacts)

    if (caller) {
      return {
        ...expandTopic(topic, contacts, getContactDetails),
        caller,
      }
    }

    return topic
  })
}
