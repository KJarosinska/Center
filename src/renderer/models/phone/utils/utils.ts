import Faker from "faker"

// TODO: remove before production
export const generateFakeData = (numberOfContacts: number) => {
  return Array(numberOfContacts)
    .fill(0)
    .map(_ => ({
      id: Faker.random.uuid(),
      firstName: Faker.name.firstName(),
      lastName: Faker.name.lastName(),
      phoneNumber: Faker.phone.phoneNumber(),
      favourite: Faker.random.boolean(),
    }))
}

export const generateSortedStructure = (fakeState: any) => {
  const alphabet = new Array(26)
    .fill(1)
    .map((_, i) => String.fromCharCode(65 + i))

  const sortedContactList = fakeState.sort((a: any, b: any) =>
    a.firstName.localeCompare(b.firstName)
  )

  const generateFakeStructure = () => {
    const fakeStructure = []

    for (const letter of alphabet) {
      fakeStructure.push({
        category: letter,
        contacts: [],
      })
    }

    fakeStructure.unshift({
      category: "Favs",
      contacts: [],
    })

    return fakeStructure
  }

  const placeContactsInStructure = () => {
    const fakeStructure: any = generateFakeStructure()
    fakeStructure.forEach((fakeContact: any) => {
      const contactLetter = fakeContact.category
      const contactList: string[] = fakeContact.contacts
      sortedContactList.forEach((sortedContact: any) => {
        if (
          sortedContact.firstName.charAt(0) === contactLetter &&
          !sortedContact.favourite
        ) {
          contactList.push(sortedContact)
        }
      })
    })

    sortedContactList.forEach((contacts: any) => {
      if (contacts.favourite) {
        fakeStructure[0].contacts.push(contacts)
      }
    })

    return fakeStructure
  }

  return placeContactsInStructure()
}

export const sanitizeContacts = (structure: any) => {
  return structure.filter((el: any) => el.contacts.length > 0)
}

export const filterContacts = (contacts: any, substring: string) => {
  return contacts.map((contactsByLetter: any) => ({
    ...contactsByLetter,
    contacts: contactsByLetter.contacts.filter((contact: any) => {
      const filterableFields = Object.values(contact).filter(value => {
        return typeof value === "string"
      })
      return filterableFields.some((value: any) =>
        value.toLowerCase().includes(substring.toLowerCase())
      )
    }),
  }))
}
