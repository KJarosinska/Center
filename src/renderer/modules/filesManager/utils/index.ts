import {
  FilesManagerData,
  FilesManagerState,
} from "Renderer/models/files-manager/files-manager.interface"

export const memoryDataWithConvertedBytes = (data: FilesManagerState) => {
  return data.memoryData.map((element: FilesManagerData) => ({
    ...element,
    occupiedMemory: element.occupiedMemory / Math.pow(1024, 3),
  }))
}
