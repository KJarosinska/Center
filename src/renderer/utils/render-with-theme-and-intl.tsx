import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import { LANGUAGE } from "Renderer/constants/languages"
import localeEn from "Renderer/locales/main/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult<Q>
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q> | Omit<RenderOptions, "queries">
) {
  return render<Q>(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={LANGUAGE.default}
        locale={LANGUAGE.default}
        messages={localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}

export function renderWithTheme(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithTheme<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult<Q>
export function renderWithTheme<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q> | Omit<RenderOptions, "queries">
) {
  const originalConsoleError = console.error
  console.error = (...args: any) => {
    if (
      !(
        args[0].startsWith("[React Intl] Cannot format message:") ||
        args[0].startsWith("[React Intl] Missing message:")
      )
    ) {
      originalConsoleError.call(console, ...args)
    }
  }
  return render<Q>(
    <ThemeProvider theme={theme}>
      <IntlProvider locale={LANGUAGE.default}>{ui}</IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
