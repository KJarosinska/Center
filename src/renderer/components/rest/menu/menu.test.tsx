import "@testing-library/jest-dom/extend-expect"
import React from "react"
import { Router } from "react-router"
import Menu from "Renderer/components/rest/menu/menu.component"
import history from "Renderer/routes/history"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"

test("matches snapshot", () => {
  const { container } = renderWithThemeAndIntl(
    <Router history={history}>
      <Menu />
    </Router>
  )
  expect(container).toMatchSnapshot()
})
