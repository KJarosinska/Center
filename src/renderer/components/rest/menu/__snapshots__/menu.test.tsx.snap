// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="sc-VigVT dCQIgv"
  >
    <test-file-stub
      classname="sc-jTzLTM cKetjh"
    />
    <div
      class="sc-gZMcBi eUufuJ"
    >
      <a
        class="sc-bwzfXH ezwVUE sc-gqjmRU cAlxsW"
        displaystyle="8"
        href="#/news"
        size="2"
      >
        <test-file-stub
          classname="sc-EHOje cmLHsF"
        />
        <p
          class="sc-bdVaJa byUSSw sc-bZQynM flUvJk"
        >
          News
        </p>
      </a>
    </div>
    <div
      class="sc-gzVnrw ipxQai"
    >
      <p
        class="sc-bdVaJa ghJYji"
      >
        Your Pure
      </p>
      <div
        class="sc-htoDjs juvQWM"
      >
        <div
          class="sc-dnqmqq iLEhhb"
        >
          <test-file-stub
            classname="sc-iwsKbI gXpsRR"
          />
        </div>
        <div
          class="sc-dnqmqq iLEhhb"
        >
          <test-file-stub
            classname="sc-iwsKbI gXpsRR"
          />
        </div>
        <div
          class="sc-dnqmqq iLEhhb"
        >
          <test-file-stub
            classname="sc-iwsKbI gXpsRR"
          />
        </div>
        <div
          class="sc-dnqmqq iLEhhb"
        >
          <test-file-stub
            classname="sc-iwsKbI gXpsRR"
          />
        </div>
      </div>
    </div>
    <div
      class="sc-gZMcBi eUufuJ"
    >
      <a
        class="sc-bwzfXH ezwVUE sc-gqjmRU cAlxsW"
        displaystyle="8"
        href="#/overview"
        size="2"
      >
        <test-file-stub
          classname="sc-EHOje cmLHsF"
        />
        <p
          class="sc-bdVaJa byUSSw sc-bZQynM flUvJk"
        >
          Overview
        </p>
      </a>
    </div>
    <div
      class="sc-gZMcBi eUufuJ"
    >
      <a
        class="sc-bwzfXH ezwVUE sc-gqjmRU cAlxsW"
        displaystyle="8"
        href="#/messages"
        size="2"
      >
        <test-file-stub
          classname="sc-EHOje cmLHsF"
        />
        <p
          class="sc-bdVaJa byUSSw sc-bZQynM flUvJk"
        >
          Messages
        </p>
      </a>
    </div>
    <div
      class="sc-gZMcBi eUufuJ"
    >
      <a
        class="sc-bwzfXH ezwVUE sc-gqjmRU cAlxsW"
        displaystyle="8"
        href="#/phone"
        size="2"
      >
        <test-file-stub
          classname="sc-EHOje cmLHsF"
        />
        <p
          class="sc-bdVaJa byUSSw sc-bZQynM flUvJk"
        >
          Phone
        </p>
      </a>
    </div>
    <div
      class="sc-gZMcBi eUufuJ"
    >
      <a
        class="sc-bwzfXH ezwVUE sc-gqjmRU cAlxsW"
        displaystyle="8"
        href="#/music"
        size="2"
      >
        <test-file-stub
          classname="sc-EHOje cmLHsF"
        />
        <p
          class="sc-bdVaJa byUSSw sc-bZQynM flUvJk"
        >
          Music
        </p>
      </a>
    </div>
    <div
      class="sc-gZMcBi eUufuJ"
    >
      <a
        class="sc-bwzfXH ezwVUE sc-gqjmRU cAlxsW"
        displaystyle="8"
        href="#/calendar"
        size="2"
      >
        <test-file-stub
          classname="sc-EHOje cmLHsF"
        />
        <p
          class="sc-bdVaJa byUSSw sc-bZQynM flUvJk"
        >
          Calendar
        </p>
      </a>
    </div>
    <div
      class="sc-gZMcBi eUufuJ"
    >
      <a
        class="sc-bwzfXH ezwVUE sc-gqjmRU cAlxsW"
        displaystyle="8"
        href="#/tools"
        size="2"
      >
        <test-file-stub
          classname="sc-EHOje cmLHsF"
        />
        <p
          class="sc-bdVaJa byUSSw sc-bZQynM flUvJk"
        >
          Tools
        </p>
      </a>
    </div>
    <div
      class="sc-gZMcBi eUufuJ"
    >
      <a
        class="sc-bwzfXH ezwVUE sc-gqjmRU cAlxsW"
        displaystyle="8"
        href="#/meditation"
        size="2"
      >
        <test-file-stub
          classname="sc-EHOje cmLHsF"
        />
        <p
          class="sc-bdVaJa byUSSw sc-bZQynM flUvJk"
        >
          Meditation
        </p>
      </a>
    </div>
    <div
      class="sc-gZMcBi eUufuJ"
    >
      <a
        class="sc-bwzfXH ezwVUE sc-gqjmRU cAlxsW"
        displaystyle="8"
        href="#/files-manager"
        size="2"
      >
        <test-file-stub
          classname="sc-EHOje cmLHsF"
        />
        <p
          class="sc-bdVaJa byUSSw sc-bZQynM flUvJk"
        >
          Files manager
        </p>
      </a>
    </div>
    <div
      class="sc-gzVnrw ipxQai"
    >
      <p
        class="sc-bdVaJa ghJYji"
      >
        Desktop App
      </p>
    </div>
    <div
      class="sc-gZMcBi eUufuJ"
    >
      <a
        class="sc-bwzfXH ezwVUE sc-gqjmRU cAlxsW"
        displaystyle="8"
        href="#/tethering"
        size="2"
      >
        <test-file-stub
          classname="sc-EHOje cmLHsF"
        />
        <p
          class="sc-bdVaJa byUSSw sc-bZQynM flUvJk"
        >
          Tethering
        </p>
      </a>
    </div>
    <div
      class="sc-gZMcBi eUufuJ"
    >
      <a
        class="sc-bwzfXH ezwVUE sc-gqjmRU cAlxsW"
        displaystyle="8"
        href="#/settings"
        size="2"
      >
        <test-file-stub
          classname="sc-EHOje cmLHsF"
        />
        <p
          class="sc-bdVaJa byUSSw sc-bZQynM flUvJk"
        >
          Settings
        </p>
      </a>
    </div>
    <div
      class="sc-gZMcBi eUufuJ"
    >
      <a
        class="sc-bwzfXH ezwVUE sc-gqjmRU cAlxsW"
        displaystyle="8"
        href="#/help"
        size="2"
      >
        <test-file-stub
          classname="sc-EHOje cmLHsF"
        />
        <p
          class="sc-bdVaJa byUSSw sc-bZQynM flUvJk"
        >
          Help
        </p>
      </a>
    </div>
  </div>
</div>
`;
